WordPress/Classes/Services/CommentService.m

```diff
@@ -992,8 +992,10 @@ - (RemoteComment *)remoteCommentWithComment:(Comment *)comment
 {
     id<CommentServiceRemote>remote;
     // TODO: refactor API creation so it's not part of the model
-    if (blog.restApi) {
-        remote = [[CommentServiceRemoteREST alloc] initWithApi:blog.restApi siteID:blog.dotComID];
+    if ([blog supports:BlogFeatureWPComRESTAPI]) {
+        if (blog.restApi) {
+            remote = [[CommentServiceRemoteREST alloc] initWithApi:blog.restApi siteID:blog.dotComID];
```

Comment:
> Couldn't the init method return nil if the API parameter was nil?

I think it's a best design to ensure you're not passing a nil attribute, but that's actually a very good point, as ServiceRemoteXMLRPC uses the inverse approach. I should change that one instead.

I know this version is more verbose right now, but if you think about the future, would you declare the initializer in Swift as `init?(api: WordPressComApi?, siteID: Int?)`?